packages/sheets-ui/src/controllers/format-painter/format-painter.controller.ts

```diff
@@ -121,18 +121,25 @@ export class FormatPainterController extends Disposable {
         const styles = workbook.getStyles();
         const stylesMatrix = new ObjectMatrix<IStyleData>();
         const merges = [];
+        let mergedMainCell: Nullable<ICellData> = null;
         for (let r = startRow; r <= endRow; r++) {
             for (let c = startColumn; c <= endColumn; c++) {
                 const cell = cellData.getValue(r, c) as ICellData;
                 stylesMatrix.setValue(r, c, styles.getStyleByCell(cell) || {});
-                const { isMergedMainCell, ...mergeInfo } = worksheet.getCellInfoInMergeData(r, c);
+                const { isMerged, isMergedMainCell, ...mergeInfo } = worksheet.getCellInfoInMergeData(r, c);
                 if (isMergedMainCell) {
+                    mergedMainCell = cell;
                     merges.push({
                         startRow: mergeInfo.startRow,
                         startColumn: mergeInfo.startColumn,
                         endRow: mergeInfo.endRow,
                         endColumn: mergeInfo.endColumn,
                     });
+                } else if (isMerged) {
+                    //see univer-pro/issues/2568
+                    if (mergedMainCell) {
```

Comment:
Could this value be `null`?

Reply:
It won't.   

PS: we found another issue relatived to this problem. This solution did not cover all case. so closed first.